src/tests/validity/generic_array_type_check.ab

```diff
@@ -0,0 +1,26 @@
+// Output
+// Generic
+// Number
+// Text
+
+fun return_generic(): [] {
+        return ["the great unknown"]
+}
+
+fun test(arr: []) {
+        if arr is [] {
+                echo "Generic"
+        }
+
+        if arr is [Num] {
+                echo "Number"
+        }
+
+        if arr is [Text] {
+                echo "Text"
+        }
+}
+
+test(return_generic())
+test([42, 69])
+test(["Hello", "Hi"])
```

Comment:
What happens when you create such array? `["hello", 32]`?

Reply:
The array construction logic has nothing to do with this PR. I checked just in case, this is the result:
```
ERROR: Expected array value of type 'Text'
```

Reply:
Add test for it

Reply:
If you want a test for it, then please add it in a different PR. The array construction logic has nothing in common with this PR.

Reply:
Yeah it's a different scope